dotcom-rendering/src/lib/scheduler.ts

```diff
@@ -0,0 +1,245 @@
+import { log as libsLog } from '@guardian/libs';
+import { measureDuration } from '../client/measureDuration';
+
+const START = performance.now();
+
+/**
+ * Keeps a count of how many tasks are currently running, so we can manage concurrency.
+ */
+let RUNNING_TASK_COUNT = 0;
+
+/**
+ * The maximum amount of concurrent tasks.
+ * @default Infinity
+ */
+let CONCURRENCY_COUNT = Infinity;
+
+/**
+ * Whether or not to log debug messages.
+ * @default process.env.NODE_ENV !== 'production'
+ */
+let DEBUG = process.env.NODE_ENV !== 'production';
+
+/**
+ * Possible task priorities.
+ *
+ * These are the Guardian's priorities, not the user's or browser's.
+ *
+ * The order of this array is important. Tasks are batched according to these
+ * priorities, and the scheduler will prefer the priority with the lowest index.
+ **/
+const PRIORITIES = ['critical', 'standard', 'optional'] as const;
```

Comment:
?
```suggestion
const PRIORITIES = ['critical', 'feature', 'enhancement'] as const;
```

Reply:
I like it, it’s a more explicit and common language!

Reply:
yeah something still not right about it for me, but it's getting there. i like that it forces you to consider what your code offers, rather than the priority you think it deserves